db_access + unittest

avigailmo opened the pull request.

ilyasoifer commented:
@avigailmo - added tests, PTAL.

avigailmo commented:
@ilyasoifer thanks!
I've added the tests but somehow they're all failed on the equals assertion although the expected and the actual dataframes look the same in all the tests.
Any chance you have an idea why?

ilyasoifer commented:
Will check. Probably some pandas nonsense

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: avigailmo ***@***.***>
Sent: Wednesday, February 19, 2025 9:12:31 AM
To: Ultimagen/ugbio-utils ***@***.***>
Cc: Ilya Soifer ***@***.***>; Mention ***@***.***>
Subject: Re: [Ultimagen/ugbio-utils] db_access + unittest (PR #105)


@ilyasoifer<https://github.com/ilyasoifer> thanks!
I've added the tests but somehow they're all failed on the equals assertion although the expected and the actual dataframes look the same in all the tests.
Any chance you have an idea why?

—
Reply to this email directly, view it on GitHub<https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGDPRCLWIAKNHCEPHRJZQFL2QQVF7AVCNFSM6AAAAABXHMOP4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGY4TKOBSHE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

[avigailmo]avigailmo left a comment (Ultimagen/ugbio-utils#105)<https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829>

@ilyasoifer<https://github.com/ilyasoifer> thanks!
I've added the tests but somehow they're all failed on the equals assertion although the expected and the actual dataframes look the same in all the tests.
Any chance you have an idea why?

—
Reply to this email directly, view it on GitHub<https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGDPRCLWIAKNHCEPHRJZQFL2QQVF7AVCNFSM6AAAAABXHMOP4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGY4TKOBSHE>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________

CONFIDENTIALITY NOTICE: This message (including any attachments) should be presumed to contain confidential, proprietary, privileged and/or private information. Information contained in this message is intended only for the recipient(s) named above. Any disclosure, reproduction, distribution or other use of this message or any attachments by any unauthorized individual or entity is strictly prohibited. If you have received this message in error, please notify the sender immediately, and delete the message and any attachments. Learn more about Ultima Genomics’ Privacy Policy<https://www.ultimagenomics.com/privacy-policy>.

avigailmo commented:
> Will check. Probably some pandas nonsense Get Outlook for Android<https://aka.ms/AAb9ysg>
> […](#)
> ________________________________ From: avigailmo ***@***.***> Sent: Wednesday, February 19, 2025 9:12:31 AM To: Ultimagen/ugbio-utils ***@***.***> Cc: Ilya Soifer ***@***.***>; Mention ***@***.***> Subject: Re: [Ultimagen/ugbio-utils] db_access + unittest (PR #105) @ilyasoifer<https://github.com/ilyasoifer> thanks! I've added the tests but somehow they're all failed on the equals assertion although the expected and the actual dataframes look the same in all the tests. Any chance you have an idea why? — Reply to this email directly, view it on GitHub<[#105 (comment)](https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGDPRCLWIAKNHCEPHRJZQFL2QQVF7AVCNFSM6AAAAABXHMOP4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGY4TKOBSHE>. You are receiving this because you were mentioned.Message ID: ***@***.***> [avigailmo]avigailmo left a comment ([Ultimagen/ugbio-utils#105](https://github.com/Ultimagen/ugbio-utils/pull/105))<[#105 (comment)](https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829)> @ilyasoifer<https://github.com/ilyasoifer> thanks! I've added the tests but somehow they're all failed on the equals assertion although the expected and the actual dataframes look the same in all the tests. Any chance you have an idea why? — Reply to this email directly, view it on GitHub<[#105 (comment)](https://github.com/Ultimagen/ugbio-utils/pull/105#issuecomment-2667695829)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AGDPRCLWIAKNHCEPHRJZQFL2QQVF7AVCNFSM6AAAAABXHMOP4GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRXGY4TKOBSHE>. You are receiving this because you were mentioned.Message ID: ***@***.***>
> ________________________________ CONFIDENTIALITY NOTICE: This message (including any attachments) should be presumed to contain confidential, proprietary, privileged and/or private information. Information contained in this message is intended only for the recipient(s) named above. Any disclosure, reproduction, distribution or other use of this message or any attachments by any unauthorized individual or entity is strictly prohibited. If you have received this message in error, please notify the sender immediately, and delete the message and any attachments. Learn more about Ultima Genomics’ Privacy Policy<https://www.ultimagenomics.com/privacy-policy>.

ok I think I found the issue 
thanks

avigailmo commented:
@ilyasoifer I was able to make 3/4 pf the tests to pass using df.astype
but test_metrics2df still fails and looks like a real diff
`                                     GcBiasSummaryMetrics  ...                         stats_coverage
                                               AT_DROPOUT  ... Non-LCR (98_Percent)___10th_percentile
                                                     self  ...                                  other
workflowId                                                 ...                                       
0440f417-be57-4887-b668-39c47cbd55aa                  NaN  ...                                    NaN
1e4440f9-76a3-4bd0-8359-8c653d5f7212                  NaN  ...                                    NaN
2989cdc2-6fa5-4931-adcf-a9c7372f162a                  NaN  ...                                    NaN
469b7436-a737-4257-9f08-7990ff95a461                  NaN  ...                                    NaN
6289c2fa-0c6a-4de8-805e-cae0e227227f                  NaN  ...                              28.805839
6674c3cc-410d-47ec-9718-b23fa34c86e1                  NaN  ...                                    NaN
9588412a-e8c2-447d-b1e7-dc46b5da3fb4                  NaN  ...                                    NaN
ddc419d2-5b94-4b92-85e0-6f17c56f3e4d             2.189859  ...                                    NaN
de06922f-07f8-4b51-843e-972308c81c6f                  NaN  ...                                    NaN
ea5e54d8-3db1-47b2-bc0e-68f38e3e89f3                  NaN  ...                                    NaN
`

Can you look?

ilyasoifer commented:
@avigailmo - I played with the tests and they seem to pass on my side. However, I must say that the test you mentioned passed even before my changes, so I hope this is sufficient

inbalzelig-ugi reviewed (approved):
Had just a small suggestion.
Can you make sure the versions are aligned?
